sample/core.py

```diff
@@ -139,7 +139,64 @@ def get_redirects(self, rdlimit="max", rdprop="pageid|title"):
         self.payload.pop('rdcontinue', None)
         self.payload['prop'] = None
         return rd_list
-    
+
+
+    def get_categoryinfo(self):
+        prop = 'categoryinfo'
+        self.payload['prop'] = prop
+
+        res = requests.get(self.base_url, params=self.payload, headers=self.headers).json()
+
+        ci_list = {}
+        for page_id, page_content in res['query']['pages'].items():
+            if prop not in page_content:
+                continue
+            ci_list[page_id] = page_content[prop]
+
+        while 'continue' in res:
+            self.payload['cicontinue'] = res['continue']['cicontinue']
+            res = requests.get(self.base_url, params=self.payload, headers=self.headers).json()
+            for page_id, page_content in res['query']['pages'].items():
+                if prop not in page_content:
+                    continue
+                if page_id not in ci_list:
+                    ci_list[page_id] = []
+                ci_list[page_id] = page_content[prop]
```

Comment:
Shouldn't you be appending the results here? Since this is inside a while loop, I am guessing assigning is not what you wanted.